pkg/openstack/loadbalancer.go

```diff
@@ -1322,6 +1327,19 @@ func (lbaas *LbaasV2) ensureOctaviaListener(lbID string, name string, curListene
 			}
 			listenerChanged = true
 		}
+		listenerKeepClientProtocol := listener.InsertHeaders[annotationXForwardedProtocol] == "true"
+		if svcConf.keepClientProtocol != listenerKeepClientProtocol {
+			updateOpts.InsertHeaders = &listener.InsertHeaders
+			if svcConf.keepClientProtocol {
+				if *updateOpts.InsertHeaders == nil {
+					*updateOpts.InsertHeaders = make(map[string]string)
+				}
+				(*updateOpts.InsertHeaders)[annotationXForwardedProtocol] = "true"
+			} else {
+				delete(*updateOpts.InsertHeaders, annotationXForwardedProtocol)
+			}
+			listenerChanged = true
+		}
```

Comment:
This completely duplicates the code above, please extract it to a function so that we can avoid repeating it.